src/pages/ReportsUpload/UploadForm.tsx

```diff
@@ -0,0 +1,313 @@
+import React, { createRef } from 'react';
+import {
+    FormikState,
+    FormikValues,
+    FormikActions,
+    FormikHandlers
+} from 'formik';
+import {
+    Form,
+    FormGroup,
+    InputGroup,
+    TextInput,
+    Button,
+    ButtonVariant,
+    Gallery,
+    GalleryItem,
+    InputGroupText,
+    Stack,
+    StackItem,
+    ActionGroup
+} from '@patternfly/react-core';
+import Dropzone from 'react-dropzone';
+
+interface UploadFormProps extends FormikState<FormikValues>, FormikActions<FormikValues>, FormikHandlers {
+    file: File;
+    onFileSelected: Function;
+    handleCancel: (e: any) => any;
+}
+
+export class UploadForm extends React.Component<UploadFormProps, { }> {
+
+    constructor(props: UploadFormProps) {
+        super(props);
+    }
+
+    componentDidMount() {
+        this.updateFieldValues();
+    }
+
+    componentDidUpdate(prevProps: any) {
+        if (prevProps.file !== this.props.file) {
+            this.updateFieldValues();
+        }
+    }
+
+    updateFieldValues() {
+        if (this.props.file) {
+            const fileName = this.props.file.name;
+            setTimeout(() => {
+                this.props.setFieldValue('file', fileName);
+                if (!this.props.values.reportName) {
+                    this.props.setFieldValue('reportName', fileName);
+                }
+            }, 0);
+        }
+    }
+
+    render() {
+        const {
+            values,
+            errors,
+            touched,
+            handleChange,
+            handleBlur,
+            handleSubmit,
+            isSubmitting
+        } = this.props;
+
+        const dropzoneRef: any = createRef();
+        const openFileDialog = () => {
+            if (dropzoneRef.current) {
+                dropzoneRef.current.open();
+            }
+        };
+
+        return (
+            <Form onSubmit={ handleSubmit }>
+                <FormGroup
+                    isRequired
+                    label="Inventory data file"
+                    fieldId="file"
+                    helperTextInvalid={ errors.file }
+                    isValid={
+                        (errors.file && touched.file) ? false : true
+                    }
+                >
+                    <Dropzone
+                        onDrop={ (files: File[]) => {
+                            this.props.onFileSelected(files);
+                        } }
+                        ref={ dropzoneRef }
+                        noClick noKeyboard
+                        multiple={ false }
+                        accept={ [ 'application/zip', 'application/json' ] }
+                    >
+                        { ({ getRootProps, getInputProps }) => {
```

Comment:
In this render function provided as children to `Dropzone`, I believe react-dropzone provides another property there called `open` that is just a function that will open the file browse dialog. So you should be able to add that to the destructuring here: `({ getRootProps, getInputProps, open }) => {` and then pass `onClick={open}` below. Then you can get rid of `dropzoneRef` and `openFileDialog`.

Reply:
You're right, I should not use `ref`. I've corrected this problem in my last commit.